p2p/src/peer.rs

```diff
@@ -197,66 +197,58 @@ impl Peer {
 
 	/// Whether the peer is considered abusive, mostly for spammy nodes
 	pub fn is_abusive(&self) -> bool {
-		if let Some(ref conn) = self.connection {
-			let conn = conn.lock();
-			let rec = conn.received_bytes.read();
-			let sent = conn.sent_bytes.read();
-			rec.count_per_min() > MAX_PEER_MSG_PER_MIN
-				|| sent.count_per_min() > MAX_PEER_MSG_PER_MIN
-		} else {
-			false
-		}
+		let conn = self.connection.lock();
+		let rec = conn.received_bytes.read();
+		let sent = conn.sent_bytes.read();
+		rec.count_per_min() > MAX_PEER_MSG_PER_MIN || sent.count_per_min() > MAX_PEER_MSG_PER_MIN
 	}
 
 	/// Number of bytes sent to the peer
 	pub fn last_min_sent_bytes(&self) -> Option<u64> {
-		if let Some(ref tracker) = self.connection {
-			let conn = tracker.lock();
-			let sent_bytes = conn.sent_bytes.read();
-			return Some(sent_bytes.bytes_per_min());
-		}
-		None
+		let conn = self.connection.lock();
+		let sent_bytes = conn.sent_bytes.read();
+		Some(sent_bytes.bytes_per_min())
 	}
 
 	/// Number of bytes received from the peer
 	pub fn last_min_received_bytes(&self) -> Option<u64> {
-		if let Some(ref tracker) = self.connection {
-			let conn = tracker.lock();
-			let received_bytes = conn.received_bytes.read();
-			return Some(received_bytes.bytes_per_min());
-		}
-		None
+		let conn = self.connection.lock();
+		let received_bytes = conn.received_bytes.read();
+		Some(received_bytes.bytes_per_min())
 	}
 
 	pub fn last_min_message_counts(&self) -> Option<(u64, u64)> {
-		if let Some(ref tracker) = self.connection {
-			let conn = tracker.lock();
-			let received_bytes = conn.received_bytes.read();
-			let sent_bytes = conn.sent_bytes.read();
-			return Some((sent_bytes.count_per_min(), received_bytes.count_per_min()));
-		}
-		None
+		let conn = self.connection.lock();
+		let received_bytes = conn.received_bytes.read();
+		let sent_bytes = conn.sent_bytes.read();
+		Some((sent_bytes.count_per_min(), received_bytes.count_per_min()))
 	}
 
 	/// Set this peer status to banned
 	pub fn set_banned(&self) {
 		*self.state.write() = State::Banned;
 	}
 
+	/// Send a msg with given msg_type to our peer via the connection.
+	fn send<T: Writeable>(&self, msg: T, msg_type: Type) -> Result<(), Error> {
+		self.connection.lock().send(msg, msg_type)
+	}
+
 	/// Send a ping to the remote peer, providing our local difficulty and
 	/// height
 	pub fn send_ping(&self, total_difficulty: Difficulty, height: u64) -> Result<(), Error> {
 		let ping_msg = Ping {
 			total_difficulty,
 			height,
 		};
-		connection!(self).send(ping_msg, msg::Type::Ping)
+		self.send(ping_msg, msg::Type::Ping)
 	}
 
 	/// Send the ban reason before banning
 	pub fn send_ban_reason(&self, ban_reason: ReasonForBan) -> Result<(), Error> {
 		let ban_reason_msg = BanReason { ban_reason };
-		connection!(self)
+		self.connection
```

Comment:
`self.send()`?

Reply:
Bad search and replace and missed these multi-line instances...